### PR TITLE
FIX-#3884: Fix read_excel() dropping empty rows

### LINKED ISSUE #3884: read_excel drops blank rows
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **Modin version** (`modin.__version__`): 0.12.1
- **Python version**: 3.10.1
- **Code we can use to reproduce**:
```
import pandas as pd
import modin.pandas as md

pd_df = pd.read_excel("testing.xlsx)
md_df = md.read_excel("testing.xlsx")
print(pd_df.shape[0]-md_df.shape[0])
```

### Describe the problem

When reading an excel file, modin drops blank rows, while pandas does not. I don't know how to prevent modin from skipping blank rows.
### Source code / logs

File to reproduce the issue:
[testing.xlsx](https://github.com/modin-project/modin/files/7776925/testing.xlsx)

### PR DESCRIPTION
**Pull request by naren-ponder**
Signed-off-by: Naren Krishna <naren@ponder.io>

## What do these changes do?

In Pandas 1.3.0, read_excel behavior changes from dropping empty rows or rows with a border in .xlsx files to including them with "NaN" values. This Modin change ensures compatibility with Pandas. Resolves #3884.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #? 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/pandas/test/test_io.py [hunk 1]
```diff
@@ -1591,6 +1591,20 @@ def test_excel_empty_line(self):
         modin_df = pd.read_excel(path)
         assert str(modin_df)
 
+    @check_file_leaks
+    def test_read_excel_empty_rows(self):
+        path = "modin/pandas/test/data/test_empty_rows.xlsx"
+        modin_df = pd.read_excel(path)
+        pandas_df = pandas.read_excel(path)
+        df_equals(modin_df, pandas_df)
```
Comment (mvashishtha) [inline modin/pandas/test/test_io.py:1599]: Why not use `eval_io` instead?

File: modin/pandas/test/test_io.py [hunk 2]
```diff
@@ -1591,6 +1591,20 @@ def test_excel_empty_line(self):
...
+
+    @check_file_leaks
+    def test_read_excel_border_rows(self):
```
Comment (mvashishtha) [inline modin/pandas/test/test_io.py:1602]: Ditto, I think you can use `eval_io`.

**Comment by mvashishtha** [comment]
Could you please add issue #3884 in the PR description?

**Review by mvashishtha — DISMISSED** [review]

File: modin/core/storage_formats/pandas/parsers.py [hunk 3]
```diff
@@ -564,12 +564,14 @@ def update_row_nums(match):
...
             usecols=usecols,
+            skip_blank_lines=False,
             **kwargs,
         )
-        # In excel if you create a row with only a border (no values), this parser will
-        # interpret that as a row of NaN values. pandas discards these values, so we
-        # also must discard these values.
-        pandas_df = parser.read().dropna(how="all")
+        pandas_df = parser.read()
+        for _ in range(pandas_df.shape[0]):
```
Comment (devin-petersohn) [inline modin/core/storage_formats/pandas/parsers.py:571]: This loop will be inefficient for even medium-sized datasets. Ideally we will not have the performance of the common cases impacted by the edge cases.

Can we `break` out of this loop in the common case (where there are no blank lines)?

**Review by devin-petersohn — COMMENTED** [review]
Glad we have a fix for this, but I worry about performance impact. Left a comment on this.

**Review by devin-petersohn — COMMENTED** [review]
Thanks @naren-ponder! Looking at this, there may be cases where the partitions are parsing both values and border-only cells. Can we create a test case that is 50 rows or so with every other row being empty with a border? This implementation might fail that test from my understanding of the code.

**Comment by naren-ponder** [comment]
> Thanks @naren-ponder! Looking at this, there may be cases where the partitions are parsing both values and border-only cells. Can we create a test case that is 50 rows or so with every other row being empty with a border? This implementation might fail that test from my understanding of the code.

Added a test case for this to `test_io.py`.

File: modin/core/storage_formats/pandas/parsers.py [hunk 4]
```diff
@@ -564,12 +564,14 @@ def update_row_nums(match):
...
             **kwargs,
         )
-        # In excel if you create a row with only a border (no values), this parser will
-        # interpret that as a row of NaN values. pandas discards these values, so we
-        # also must discard these values.
-        pandas_df = parser.read().dropna(how="all")
+        pandas_df = parser.read()
+        check_nan_loc = pandas_df.isnull()[::-1].all(axis=1)
+        if check_nan_loc.all() and check_nan_loc.shape[0] > 1:
+            pandas_df = pandas_df.drop(pandas_df.index[: pandas_df.shape[0] + 1])
```
Comment (dchigarev) [inline modin/core/storage_formats/pandas/parsers.py:573]: This operation seems to leave the frame empty as it drops all of the rows, is this intended? If so, then I would suggest an easier way to create an empty frame:
```suggestion
            pandas_df = pandas.DataFrame(columns=pandas_df.columns)
```

File: modin/core/storage_formats/pandas/parsers.py [hunk 5]
```diff
@@ -564,12 +564,14 @@ def update_row_nums(match):
...
+            skip_blank_lines=False,
             **kwargs,
         )
-        # In excel if you create a row with only a border (no values), this parser will
-        # interpret that as a row of NaN values. pandas discards these values, so we
-        # also must discard these values.
-        pandas_df = parser.read().dropna(how="all")
+        pandas_df = parser.read()
+        check_nan_loc = pandas_df.isnull()[::-1].all(axis=1)
+        if check_nan_loc.all() and check_nan_loc.shape[0] > 1:
```
Comment (dchigarev) [inline modin/core/storage_formats/pandas/parsers.py:572]: why do we need to reverse `isnull()` result?
```suggestion
        if len(pandas_df) > 1 and pandas_df.isnull().all().all():
```

File: modin/core/storage_formats/pandas/parsers.py [hunk 6]
```diff
@@ -564,12 +564,13 @@ def update_row_nums(match):
...
             usecols=usecols,
+            skip_blank_lines=False,
             **kwargs,
         )
-        # In excel if you create a row with only a border (no values), this parser will
-        # interpret that as a row of NaN values. pandas discards these values, so we
-        # also must discard these values.
-        pandas_df = parser.read().dropna(how="all")
+        pandas_df = parser.read()
+        if len(pandas_df) > 1 and pandas_df.isnull().all().all():
```
[the remark this answers is not in the document]
Comment (naren-ponder) [reply modin/core/storage_formats/pandas/parsers.py:571]: That still works because of the return of `parser.read()` and this `if` case does not get hit.

Comment (mvashishtha) [inline modin/core/storage_formats/pandas/parsers.py:571]: Could you please add a comment explaining why we create a new pandas DataFrame if this test is true?

File: modin/pandas/test/test_io.py [hunk 7]
```diff
@@ -1592,6 +1592,30 @@ def test_excel_empty_line(self):
...
+            fn_name="read_excel",
+            io="modin/pandas/test/data/test_border_rows.xlsx",
+        )
+
+    @check_file_leaks
+    def test_read_excel_every_other_nan(self):
+        # Test for reading excel dataframe with every other row as a NaN value
+        eval_io(
+            fn_name="read_excel",
+            io="modin/pandas/test/data/test_every_other.xlsx",
```
Comment (mvashishtha) [inline modin/pandas/test/test_io.py:1616]: I think it will be clearer in the future if this is called `every_other_row_nan.xlsx`

**Review by jeffreykennethli — APPROVED** [review]

**Review by devin-petersohn — APPROVED** [review]
Thanks @naren-ponder, LGTM!

**Pull request merged**